x-pack/plugins/cases/common/types/api/case/v1.ts

```diff
@@ -62,10 +64,17 @@ const CaseCustomFieldNumberWithValidationRt = rt.strict({
   value: rt.union([CaseCustomFieldNumberWithValidationValueRt({ fieldName: 'value' }), rt.null]),
 });
 
+const CaseCustomFieldListWithValidationRt = rt.strict({
+  key: rt.string,
+  type: CustomFieldListTypeRt,
+  value: rt.union([CaseCustomFieldListWithValidationValueRt('value'), rt.null]),
```

Comment:
I am not sure this is the right approach. Why not just save the key?

Testing this with the API led me to some weird scenarios.

Try the following:
1. Create a **list** custom field with a few different options.
2. Send a case POST request where the custom field value has:(1) a correct `key`, and(2) a **wrong** `value/label`.

The case detail page renders the correct label but the case GET API returns the wrong label that does not exist in the configuration.

Some screenshots illustrating this scenario:

### My configuration
<img width="844" alt="Screenshot 2024-12-04 at 15 45 39" src="https://github.com/user-attachments/assets/437c5a8a-36e7-44a0-a6c7-0edbb0c0646c">

### The POST request
<img width="1341" alt="Screenshot 2024-12-04 at 15 47 12" src="https://github.com/user-attachments/assets/d28582a1-b537-4689-b28b-9989a302f271">

### The case detail page
<img width="1712" alt="Screenshot 2024-12-04 at 15 48 12" src="https://github.com/user-attachments/assets/f33d92bf-542f-412d-a7af-91181fc78967">

Reply:
> I am not sure this is the right approach. Why not just save the key?

Already decided on in https://github.com/elastic/kibana/issues/168378

Text of the custom field labels need to be searchable and queryable

Reply:
> Send a case POST request where the custom field value has:(1) a correct key, and(2) a wrong value/label.

Let me take a look and see if I can get the POST to only contain the key and derive the label from it on the server side. That's probably the safest.